perun-core/src/main/java/cz/metacentrum/perun/core/impl/MembersManagerImpl.java

```diff
@@ -284,6 +301,15 @@ public void removeSponsor(PerunSession session, Member sponsoredMember, User spo
 		}
 	}
 
+	@Override
+	public void deleteAllSponsors(PerunSession session, Member sponsoredMember) throws InternalErrorException {
+		try {
+			jdbc.update("DELETE FROM members_sponsored WHERE sponsored_id=?", sponsoredMember.getId() );
```

Comment:
This bulk deletion of all sponsors in the database will not generate needed auditlog messages, so I think that this method should not be implemented on the database level, it should be implemented on the service level where it will find all sponsors and remove them one by one using the method MembesManagerBl.removeSponsor(), so that an auditlog message will be generated for each one and expiration would be set correctly.

Reply:
This method is only in "impl" layer and need to be used with caution. You can see it is used only in method unsetSponsorshipForMember and this method takes care of all necessary messages.
Also we don't want to change expiration when removing sponsored state of a Member.